.github/workflows/build.yml

```diff
@@ -119,6 +119,55 @@ jobs:
       - name: Test
         run: cd build && ctest -C Release -j3 --output-on-failure
 
+  freebsd:
+    name: FreeBSD
+    runs-on: ubuntu-latest
+    if: github.event_name != "pull_request"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: true
+
+      - name: Prepare
+        run: ./autogen.sh
+
+      - name: Build & test
+        uses: vmactions/freebsd-vm@debf37ca7b7fa40e19c542ef7ba30d6054a706a4 # v1.1.5
+        with:
+          usesh: true
+          run: |
+            set -e
+            ./configure --enable-jit --enable-pcre2-16 --enable-pcre2-32
+            make -j3 CPPFLAGS='-Wall -Wextra -Werror'
+            make check
+
+  solaris:
+    name: Solaris
+    runs-on: ubuntu-latest
+    if: github.event_name != "pull_request"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: true
+
+      - name: Prepare
+        run: ./autogen.sh
+
+      - name: Build & test
+        uses: vmactions/solaris-vm@a89b9438868c70db27e41625f0a5de6ff5e90809 # v1.1.0
+        with:
+          usesh: true
+          prepare: |
+            pkg install developer/gcc system/header
+
+          run: |
+            set -e
+            ./configure --enable-jit --enable-pcre2-16 --enable-pcre2-32
+            gmake -j3 CPPFLAGS='-Wall -Wextra -Werror'
```

Comment:
It failed with Solaris `make`, so I just used gmake. I didn't consider it important what make system works - as long as the code compiles & runs, we're fine.

But if you prefer using Solaris make, I'll try and fix it.

I've also decided to have an attempt at using Solaris `cc` rather than GCC.